src/inference/src/dev/core_impl.cpp

```diff
@@ -979,84 +927,199 @@ void ov::CoreImpl::AddExtensionUnsafe(const InferenceEngine::IExtensionPtr& exte
+    if (config_name == ov::core_property_keys.name()) {
+        std::set<std::string> res;
+        for (auto& it : _core_plugin_properties._properties) {
+            res.emplace(it.first);
+        }
+        return res;
+    }
+
+    IE_THROW() << "Exception: unsupported core property name: '" << config_name << "'";
 }
 
-void ov::CoreImpl::CoreConfig::set_cache_dir_for_device(const std::string& dir, const std::string& name) {
-    std::lock_guard<std::mutex> lock(_cacheConfigMutex);
-    fill_config(_cacheConfigPerDevice[name], dir);
+bool ov::CoreImpl::CoreConfig::is_core_config(const std::string& config_name) const {
+    auto ret = _core_plugin_properties._properties.find(config_name) != _core_plugin_properties._properties.end() ||
+               _core_global_properties._properties.find(config_name) != _core_global_properties._properties.end() ||
+               config_name == ov::core_property_keys.name();
+    return ret;
 }
 
-std::string ov::CoreImpl::CoreConfig::get_cache_dir() const {
-    std::lock_guard<std::mutex> lock(_cacheConfigMutex);
-    return _cacheConfig._cacheDir;
+bool ov::CoreImpl::CoreConfig::core_config_supported(const ov::Plugin& plugin, const std::string& config_name) const {
+    auto supported = false;
+    auto device_name = plugin.get_name();
+
+    if (_core_plugin_properties._properties.find(config_name) != _core_plugin_properties._properties.end()) {
+        supported = device_name.find("CPU") != std::string::npos || device_name.find("GPU") != std::string::npos ||
+                    device_name.find("GNA") != std::string::npos || device_name.find("AUTO") != std::string::npos ||
+                    device_name.find("MULTI") != std::string::npos || device_name.find("BATCH") != std::string::npos ||
+                    device_name.find("HETERO") != std::string::npos;
+        // Need query for external plugins: ARM, CUDA, VPUX...
+        if (!supported) {
+            supported = util::contains(plugin.get_property(ov::supported_properties), config_name);
+        }
+    }
+    return supported;
+}
+
+void ov::CoreImpl::CoreConfig::remove_core_config(const ov::Plugin& plugin,
+                                                  ov::AnyMap& config,
+                                                  const bool update) const {
+    std::lock_guard<std::mutex> lock(_CoreConfigCacheMutex);
+    for (auto& it : _core_plugin_properties._properties) {
+        auto item = config.find(it.first);
+        if (item != config.end()) {
+            auto key = item->first;
+            auto value = item->second.as<std::string>();
+            if (update && !value.empty()) {
+                auto device_name = plugin.get_name();
+                fill_config(_core_properties_cache_per_device[device_name], key, value);
+            }
+            config.erase(item);
+        }
+    }
+}
+
+template <typename T>
+void ov::CoreImpl::CoreConfig::update_input_config(ov::Plugin& plugin, std::map<std::string, T>& config) const {
```

Comment:
Looks like you don't change plugin
```suggestion
void ov::CoreImpl::CoreConfig::update_input_config(const ov::Plugin& plugin, std::map<std::string, T>& config) const {
```

Reply:
Maybe can we support only ov::AnyMap as a config?

Reply:
More details?
Do you mean use `ov::AnyMap` replace `std::map<std::string, T>` here?

Reply:
Yes

Reply:
ok, update it